rust/notes.md

```diff
@@ -0,0 +1,54 @@
+# Implementors' notes
+
+## Desired subjects
+
+This is a more full list of topics to cover than
+in the README.
+
+- error handling
+  - simple vs. complex, Fail vs StdError, etc
+- logging w/ log and slog
+  - how env_logger works?
+- trees vs maps
+- async vs sync networking
+- sync file io and solutions to blocking
+- buffered vs unbuffered i/o
+- benchmarking, criterion and critcmp, black_box
+- RUST_BACKTRACE
+- how to find crates
+- where to ask questions
+- futures
+- tokio
+- async/await? - maybe future iterations
+- generic placeholder idiom - let foo: Vec<_> =
+- construction w/ iterators idiom
+- semver trick
+- impl trait, the Into<Option<_>> trick
+- Rust history, culture, design principles
+- rustfmt, clippy, configuring both
+- rust 2018?
```

Comment:
Given that this is the current and stable edition, yes. I think covering *2015* is optional beyond a small mention.